### PR TITLE
cairo 2.9.0-dev (native v0.2.3 branch)

### LINKED ISSUE #879: [2.9.0] Implement `int_range_try_new` and `int_range_pop_front` libfuncs
The target branch for the pr should be https://github.com/lambdaclass/cairo_native/pull/877

### PR DESCRIPTION
**Pull request by edg-l**
Fixes #879

This is cairo-native with cairo 2.9.0 features

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/values.rs [hunk 1]
```diff
@@ -76,6 +76,10 @@ pub enum Value {
         #[serde(with = "range_serde")]
         range: Range,
     },
+    IntRange {
+        x: Box<Value>,
+        y: Box<Value>,
+    },
```
Comment (azteca1998) [inline src/values.rs:82]: This avoids an allocation.
```suggestion
    IntRange(Box<Range<Value>>),
```

File: src/values.rs [hunk 2]
```diff
@@ -511,6 +515,36 @@ impl Value {
...
+
+                        let ptr = arena.alloc_layout(struct_layout.pad_to_align()).as_ptr();
+
+                        for (layout, offset, member_ptr) in data {
+                            std::ptr::copy_nonoverlapping(
+                                member_ptr.cast::<u8>().as_ptr(),
+                                ptr.byte_add(offset),
+                                layout.size(),
+                            );
+                        }
```
Comment (azteca1998) [inline src/values.rs:541]: Why not just write it directly instead of creating a vec, pushing two items and looping over it?

File: src/values.rs [hunk 3]
```diff
@@ -802,6 +836,35 @@ impl Value {
...
+                            None => (member_layout, 0),
+                        };
+                        layout = Some(new_layout);
+
+                        members.push(Self::from_ptr(
+                            NonNull::new(ptr.as_ptr().byte_add(offset)).unwrap(),
+                            member_ty,
+                            registry,
+                        )?);
+                    }
```
Comment (azteca1998) [inline src/values.rs:858]: Same as before, but in reverse.

File: scripts/bench-hyperfine.sh [hunk 4]
```diff
@@ -57,7 +57,6 @@ run_bench() {
         -L"$OUTPUT_DIR/" \
         -Wl,-rpath "$MLIR_DIR/lib" \
         -Wl,-rpath "$OUTPUT_DIR" \
-        -Wl,--rpath-link "$OUTPUT_DIR" \
```
Comment (JulianGCalderon) [inline scripts/bench-hyperfine.sh:60]: Out of curiosity, why is this argument unnecessary?

Comment (edg-l) [reply scripts/bench-hyperfine.sh:60]: it wasnt necessary, i added it somewhen trying some stuff, but this argument doesnt exist on macos

**Review by azteca1998 — APPROVED** [review]

**Review by FrancoGiachetta — APPROVED** [review]

**Pull request merged**